src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -32,7 +32,8 @@ import com.databricks.spark.csv.util._
 import com.databricks.spark.sql.readers._
 
 case class CsvRelation protected[spark] (
-    location: String,
+    baseRDD: () => RDD[String],
```

Comment:
With this change, we don't need to pass charset to CsvRelation any more. You can remove it from list of arguments.